modules/cudf/src/table/copying.cpp

```diff
@@ -28,4 +29,105 @@ ObjectUnwrap<Table> Table::gather(Column const& gather_map,
   return Table::New(cudf::gather(cudf::table_view{{*this}}, gather_map, bounds_policy, mr));
 }
 
+ObjectUnwrap<Table> Table::scatter_scalar(
+  std::vector<std::reference_wrapper<const cudf::scalar>> const& source,
+  Column const& indices,
+  bool check_bounds,
+  rmm::mr::device_memory_resource* mr) const {
+  try {
+    return Table::New(cudf::scatter(source, indices.view(), this->view(), check_bounds, mr));
+  } catch (cudf::logic_error const& err) { NAPI_THROW(Napi::Error::New(Env(), err.what())); }
+}
+
+ObjectUnwrap<Table> Table::scatter_table(Table const& source,
+                                         Column const& indices,
+                                         bool check_bounds,
+                                         rmm::mr::device_memory_resource* mr) const {
+  try {
+    return Table::New(cudf::scatter(source.view(), indices.view(), this->view(), check_bounds, mr));
+  } catch (cudf::logic_error const& err) { NAPI_THROW(Napi::Error::New(Env(), err.what())); }
+}
+
+Napi::Value Table::gather(Napi::CallbackInfo const& info) {
+  CallbackArgs args{info};
+  if (!Column::is_instance(args[0])) {
+    throw Napi::Error::New(info.Env(), "gather selection argument expects a Column");
+  }
+  auto& selection = *Column::Unwrap(args[0]);
+  if (selection.type().id() == cudf::type_id::BOOL8) {
+    return this->apply_boolean_mask(selection)->Value();
+  }
+  return this->gather(selection)->Value();
+}
+
+Napi::Value Table::scatter_scalar(Napi::CallbackInfo const& info) {
+  CallbackArgs args{info};
+  if (!args[0].IsArray()) {
+    throw Napi::Error::New(info.Env(), "scatter_scalar source argument expects an array");
+  }
+  auto const source_array = args[0].As<Napi::Array>();
+  std::vector<std::reference_wrapper<const cudf::scalar>> source{};
+
+  for (uint32_t i = 0; i < source_array.Length(); ++i) {
+    if (!Scalar::is_instance(source_array.Get(i))) {
+      throw Napi::Error::New(info.Env(),
+                             "scatter_scalar source argument expects an array of scalars");
+    }
+    auto& scalar = *Scalar::Unwrap(source_array.Get(i).ToObject());
+    source.push_back(scalar);
+  }
+
+  if (!Column::is_instance(args[1])) {
+    throw Napi::Error::New(info.Env(), "scatter_scalar indices argument expects a Column");
+  }
+  auto& indices = *Column::Unwrap(args[1]);
+
+  if (args.Length() == 2 or (args.Length() >= 2 and args[2].IsUndefined())) {
+    return scatter_scalar(source, indices)->Value();
+  }
+
+  if (args.Length() > 2 and args[2].IsBoolean()) {
+    auto check_bounds = NapiToCPP(args[2]);
+    if (args.Length() == 3) {
+      return scatter_scalar(source, indices, check_bounds)->Value();
+    } else if (args.Length() == 4 and args[2].IsBoolean()) {
+      auto* mr = NapiToCPP(info[3]).operator rmm::mr::device_memory_resource*();
+      return scatter_scalar(source, indices, check_bounds, mr)->Value();
+    }
+  }
+  NAPI_THROW(Napi::Error::New(Env(),
```

Comment:
Should probably use the `env` of the caller rather than the `env` of the Table instance.
```suggestion
  NAPI_THROW(Napi::Error::New(info.Env(),
```